perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_resource_attribute_def_def_filesQuota.java

```diff
@@ -48,30 +54,30 @@ public void checkAttributeSemantics(PerunSessionImpl perunSession, Member member
 
 		//Get FilesQuota value
 		if(attribute.getValue() != null) {
-			filesQuota = (Integer) attribute.getValue();
+			filesQuota = attribute.valueAsInteger();
 		} else {
 			try {
 				attribute = perunSession.getPerunBl().getAttributesManagerBl().getAttribute(perunSession, resource, A_R_defaultFilesQuota);
 			} catch (AttributeNotExistsException ex) {
 				throw new ConsistencyErrorException("Attribute with defaultFilesQuota from resource " + resource.getId() + " could not obtained.", ex);
 			}
 			if(attribute != null && attribute.getValue() != null) {
-				filesQuota = (Integer) attribute.getValue();
+				filesQuota = attribute.valueAsInteger();
+				if(filesQuota < 0) throw new WrongReferenceAttributeValueException(attribute, null, resource, member, attribute + " cannot be less than 0.");
```

Comment:
Same here.

Reply:
The reason for this check is that _attribute_ does not necessarily hold input attribute but it can hold _defaultFilesLimit_ attribute unlike in syntax check where it's definitely attribute given by the method. Should I change the check to check only if it's _defaultFilesLimit_?

Reply:
Sorry I didn't see that. You don't have to change it